packages/admin/src/pages/super-admin-dashboard/manage-departments/index.page.tsx

```diff
@@ -13,13 +13,19 @@ import { GetServerSidePropsContext } from 'next';
 import { Department } from '../types';
 import { Row } from 'gap-web-ui/dist/cjs/components/summary-list/SummaryList';
 import styles from './manage-departments.module.scss';
+import { AxiosError } from 'axios';
+import { handleAxiosError } from '../../../utils/handleAxiosError';
 
 export async function getServerSideProps(context: GetServerSidePropsContext) {
   const fetchPageData = async (jwt: string) => ({
     departments: await getAllDepartments(jwt),
     userId: context.query?.userId || '',
   });
-  return { props: await fetchPageData(getUserTokenFromCookies(context.req)) };
+  try {
+    return { props: await fetchPageData(getUserTokenFromCookies(context.req)) };
+  } catch (err) {
+    return handleAxiosError(err as AxiosError);
+  }
```

Comment:
Could abstract this logic into the new function?

Would need to accept a function as a param, generically type it, and return either the functions response, or if there's an error the logic you already have there